zerver/tests/test_messages.py

```diff
@@ -233,21 +234,21 @@ def assert_invalid_email() -> Any:
         self.send_personal_message(user1_email, feedback_email, sender_realm="1.example.com")
         assert_message_received(feedback_bot, user1)
 
-        # Users can PM cross-realm bots on non-zulip realms.
+        # Users cannot PM cross-realm users on non-zulip realms currently.
```

Comment:
Is this new comment right? I think this bit of test code fails for the simpler reason that the test no longer creates a user with email `support_email`, because it's no longer possible to create a user in the normal way if it has a cross-realm bot's email.

Maybe the check when creating a new user should become that *either* the email doesn't yet exist, or it's not a system bot? Then I think this test wouldn't have to change at all -- I think all the changes are fallout from the test's setup no longer working right.

Reply:
Yeah, the changes to this test seem questionable.  I think the check you describe sounds correct.

Reply:
Hmm, actually trying it, the test runs just fine without even making that change to the check.

There is some error-logging spew from welcome-bot trying to send PMs, which is fixed by adding it to the overridden `CROSS_REALM_BOT_EMAILS` -- but that exists in master. Presumably caused by welcome-bot being created, and added to the normal `CROSS_REALM_BOT_EMAILS`, and this overridden version lagging behind.

I am a little puzzled how... aha. The new validator is only being used on realm creation, not on registration in an existing realm. That would be good to add a test for too. :-)